crates/sui-indexer-alt/src/handlers/kv_epoch_starts.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use anyhow::{bail, Context, Result};
+use diesel_async::RunQueryDsl;
+use sui_types::{
+    full_checkpoint_content::CheckpointData,
+    sui_system_state::{get_sui_system_state, SuiSystemStateTrait},
+    transaction::{TransactionDataAPI, TransactionKind},
+};
+
+use crate::{
+    db, models::epochs::StoredEpochStart, pipeline::concurrent::Handler, pipeline::Processor,
+    schema::kv_epoch_starts,
+};
+
+pub struct KvEpochStarts;
+
+impl Processor for KvEpochStarts {
+    const NAME: &'static str = "kv_epoch_starts";
+
+    type Value = StoredEpochStart;
+
+    fn process(&self, checkpoint: &Arc<CheckpointData>) -> Result<Vec<Self::Value>> {
+        let CheckpointData {
+            checkpoint_summary,
+            transactions,
+            ..
+        } = checkpoint.as_ref();
+
+        // If this is the last checkpoint in the current epoch, it will contain enough information
+        // about the start of the next epoch.
+        if !checkpoint_summary.is_last_checkpoint_of_epoch() {
+            return Ok(vec![]);
+        }
```

Comment:
might be misunderstanding how this works but, iiuc this checkpoint summary would also contain end_of_eopch data? If so, then addressing your response to another comment, at this checkpoint summary the network_total_transactions can serve as the tx_lo

Reply:
I think the issue here is that this is the `kv_epoch_starts` pipeline, and the earlier question was about `kv_epoch_ends` -- the two pipelines are intentionally kept distinct and append-only.

I think to make this suggestion work, you would need to have the end pipeline write the tx_lo for the start pipeline which goes against that pattern (and of course, you can fetch tx_lo as tx_hi from the previous epoch).